src/sentry/static/sentry/app/stores/groupingStore.jsx

```diff
@@ -0,0 +1,368 @@
+import Reflux from 'reflux';
+
+import IndicatorStore from './indicatorStore';
+import {Client} from '../api';
+import GroupingActions from '../actions/groupingActions';
+
+const api = new Client();
+
+// Between 0-100
+const SIMILARITY_THRESHOLD = 50;
+
+// @param score: {[key: string]: number}
+const getAvgScore = score => {
+  let scoreKeys = (score && Object.keys(score)) || [];
+  return Math.round(
+    scoreKeys.map(key => score[key]).reduce((acc, s) => acc + s * 100, 0) /
+      scoreKeys.length
+  );
+};
+
+const GroupingStore = Reflux.createStore({
+  listenables: [GroupingActions],
+  init() {
+    let state = this.getDefaultState();
+
+    Object.entries(state).forEach(([key, value]) => {
+      this[key] = value;
+    });
+  },
+
+  getDefaultState() {
+    return {
+      mergedItems: [],
+      unmergeList: new Set(),
+      unmergeState: new Map(),
+      unmergeDisabled: false,
+
+      similarItems: [],
+      filteredSimilarItems: [],
+      similarLinks: '',
+      mergeState: new Map(),
+      mergeList: new Set(),
+      mergedLinks: '',
+      mergeDisabled: false,
+
+      loading: true,
+      error: false
+    };
+  },
+
+  setStateForId(map, id, newState) {
+    let state = (map.has(id) && map.get(id)) || {};
+    let mergedState = Object.assign({}, state, newState);
+    map.set(id, mergedState);
+    return mergedState;
+  },
+
+  // Fetches data
+  onFetch(toFetchArray) {
```

Comment:
I'm not seeing the advantage of having this extra level of indirection. The action creators are coupled to one set of actions, wouldn't it be the same if they were implemented on the store?

That's not to say this shouldn't be refactored, but maybe as different action handlers `onFetch`, `onFetchStart`, `onFetchSuccess`, `onFetchError`.

Should the API calls to merge/unmerge also be action creators?